demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +362,423 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+                        AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT, HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Retrieve the path and length from AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT. */
+        httpStatus = getUrlPath( AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT,
+                                 sizeof( AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT ) - 1,
+                                 &pPath,
+                                 &pathLen );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to get path from URL %s: Error=%s.",
+                        AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT, HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    /* Initialize Request header buffer. */
+    requestHeaders.pBuffer = response->pBuffer;
+    requestHeaders.bufferLen = response->bufferLen;
+
+    /* Set HTTP request parameters to get temporary AWS IOT credentials. */
+    requestInfo.pMethod = HTTP_METHOD_GET;
+    requestInfo.methodLen = sizeof( HTTP_METHOD_GET ) - 1;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = pathLen;
+    requestInfo.pHost = pAddress;
+    requestInfo.hostLen = addressLen;
+    requestInfo.reqFlags = 0;
+
+    response->pHeaderParsingCallback = NULL;
+
+    if( returnStatus == true )
+    {
+        /* Initialize request headers. */
+        httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to initialize request headers: Error=%s.",
+                        HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Add AWS_IOT_THING_NAME_HEADER_FIELD header to the HTTP request headers. */
+        httpStatus = HTTPClient_AddHeader( &requestHeaders,
+                                           AWS_IOT_THING_NAME_HEADER_FIELD,
+                                           sizeof( AWS_IOT_THING_NAME_HEADER_FIELD ) - 1,
+                                           AWS_IOT_THING_NAME,
+                                           sizeof( AWS_IOT_THING_NAME ) );
+
+        if( httpStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to add x-amz-iot-thing-name header to request headers: Error=%s.",
+                        HTTPClient_strerror( httpStatus ) ) );
+            returnStatus = false;
+        }
+    }
+
+    if( returnStatus == true )
+    {
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( transportInterface,
+                                      &requestHeaders,
+                                      NULL,
+                                      0,
+                                      response, 0 );
+
```

Comment:
Please provide in-line comments about what the `0` magic numbers mean. If one of them is a timeout value, why is that provided

Reply:
These are the the numbers which are used if we have some content in the request body, but as for HTTP S3 demos we are not having anything as a part of request body, so 0 is passed. I can mention this in comment.